core/src/test/java/org/apache/calcite/rel/RelCollationTest.java

```diff
@@ -98,6 +101,12 @@
     assertThat(RelCollations.containsOrderless(collations, Arrays.asList(0)), is(false));
   }
 
+  @Test void testNumDistinctVals() {
```

Comment:
The test is not relevant to collation so it doesn't fit well in this class. It may be better to create another class `RelMdDistinctRowCountTest` or `RelMdUtilTest` (depending on what exactly you are going to test) and put it inside. 

Also it may be a good opportunity to add a bit more tests around this metric (big/small domain, selected combinations and boundary cases).

Reply:
Thanks a lot for the good suggestion.
I have moved it to `RelMdUtilTest`, and added more test cases for various conditions.